### PR TITLE
Document how to configure the underlying http client. close #532

### LINKED ISSUE #532: Document how to configure the underlying http client
such as increasing the connection timeout in AHC

### PR DESCRIPTION
**Pull request by schleichardt**

### CODE REVIEW

**Comment by schleichardt** [comment]
@lauraluiz please review

preview is on http://sphereio.github.io/sphere-jvm-sdk/javadoc/documentation/532/configure-underlying-http-client/io/sphere/sdk/meta/SphereClientTuningDocumentation.html#configure-underlying-http-client

File: sphere-java-client-ahc-1_9/src/it/java/io/sphere/sdk/client/CustomClientConfigDemoTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+public class CustomClientConfigDemoTest {
+    /**
+     * Creates a {@link HttpClient} that can be used by the {@link SphereClient}.
+     * @return new http client with custom settings
+     */
+    public static HttpClient createCustomHttpClient() {
+        final AsyncHttpClientConfig httpClientConfig = new AsyncHttpClientConfig.Builder()
+                .setEnabledProtocols(new String[]{"TLSv1.2"})//required
+                //some examples what can be configured
```
Comment (lauraluiz) [inline sphere-java-client-ahc-1_9/src/it/java/io/sphere/sdk/client/CustomClientConfigDemoTest.java]: `some examples that can be configured` or `some examples can be configured`?

File: sphere-java-client-ahc-1_9/src/it/java/io/sphere/sdk/client/CustomClientConfigDemoTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+//                .setProxyServer(proxy)
+                .build();
+        final AsyncHttpClient asyncHttpClient = new AsyncHttpClient(httpClientConfig);
+        return AsyncHttpClientAdapter.of(asyncHttpClient);
+    }
+
+    /**
+     * Shows the initialization of a {@link SphereClient} with a custom {@link HttpClient} provider.
+     */
+    private void demoCreateClient() {
```
Comment (lauraluiz) [inline sphere-java-client-ahc-1_9/src/it/java/io/sphere/sdk/client/CustomClientConfigDemoTest.java]: Is this method related to the other method or the test? Because I guess it's just to show how to use it in the sphere client, but that confuses a bit, I think. I cannot think of any suggestion to improve it though, sorry.

Comment (schleichardt) [inline sphere-java-client-ahc-1_9/src/it/java/io/sphere/sdk/client/CustomClientConfigDemoTest.java]: At some point Stefan an Markus will have a look. ;)

**Comment by schleichardt** [comment]
@lauraluiz can I merge?

**Comment by lauraluiz** [comment]
Yep
